core/Piranha.Azure.BlobStorage/BlobStorage.cs

```diff
@@ -66,31 +61,18 @@ public class BlobStorage : IStorage
             string containerName = "uploads",
             BlobStorageNaming naming = BlobStorageNaming.UniqueFileNames)
         {
-            _storage = CloudStorageAccount.Parse(connectionString);
-            _containerName = containerName;
+            _storage = new BlobContainerClient(connectionString, containerName);
             _naming = naming;
+            _storage.CreateIfNotExists(PublicAccessType.Blob);
```

Comment:
Does this create a request towards Azure? If so I would recommend keeping the async variant as was before.

Reply:
@mikaellindemann this should not be inside the constructor, agree. On the other hand, the method `IStorage.OpenAsync()` would not be a proper place either. Maybe an idea to add a method `IStorage.InitializeAsync()` where initial plumbing and checks can be performed?

Reply:
@mikaellindemann this now moved to:
https://github.com/PiranhaCMS/piranha.core/pull/1564/files#diff-1d62fb7d41884231bb1421e147495f8cd812a633cde9981b64684e29639056c4R164

Reply:
My point was that if it does asynchronous operations (IO being one of them) then it should be an asynchronous function. I didn't check what is behind CreateIfNotExists, but since ExistsAsync and CreateAsync was in use before, I guess that CreateIfNotExists is waiting on the asynchronous operation behind the scenes.

Not sure if I encouraged this change you have made, sorry if I did, but now you have introduced a breaking change in the storage extension point of Piranha. Also, you have limited any storage implementation not to be session based.

Wouldn't it make sense to keep the storage and session implementation as was, such that a BlobContainerClient was the underlying session for a BlobStorage? (Tbh. I don't know the details of IStorage, IStorageSession and Azure BlobStorage, so maybe this is a bad suggestion).

Reply:
@mikaellindemann I reversed most of those breaking changes. Guess I was a bit too enthousiastic. Please take a second look at the result.

Reply:
And with regards to that CreateIfNotExists, it's about ensuring the parent Storage Container is available. So this should optimally be part of the App Init chaining.

Reply:
And with Azure.Storage.Blobs assembly there is no real concept of sessions.
The `BlobContainerClient` is maintained as a singleton via DI / `BlobStorage` class. And `BlobClient` class instances are lightweight and are instantiated from the `BlobContainerClient`.

Reply:
@mikaellindemann I have now reverted any breaking changes to existing interfaces. And reverted all changes to the `Piranha.Local.FileStorage` source.
What remains now inside this PR:

1. the move to the Azure SDK v.12 using the recommended `Azure.Storage.Blobs` Nuget packages
2. Add the interface `IInitializable` to optionally indicate that a class implements `Init()` method, which is then called via App.cs